source/NVDAObjects/UIA/__init__.py

```diff
@@ -1469,6 +1469,10 @@ def event_UIA_notification(self, notificationKind=None, notificationProcessing=N
 		if self.appModule != api.getFocusObject().appModule:
 			return
 		if displayString:
+			if notificationProcessing in (UIAHandler.NotificationProcessing_ImportantMostRecent, UIAHandler.NotificationProcessing_MostRecent):
+				# These notifications superseed earlier notifications.
+				# Note that no distinction is made between important and non-important.

```

Comment:
With the merging of #7599 we should now be able to distinguish between important and non-important notifications (possibly with speech priorities) instead of cancelling speech outright. Is this worth doing?